lib/shopify_transporter/exporters/magento/product_exporter.rb

```diff
@@ -62,20 +63,45 @@ def merge_simple_product_with_parent(product, product_mappings)
         def with_attributes(product)
           product_with_base_attributes = product
             .merge(images: images_attribute(product[:product_id]))
-            .merge(info_for(product[:product_id]))
+            .merge(info_for(product))
             .merge(tags: product_tags(product[:product_id]))
 
           case product[:type]
           when 'simple'
-            product_with_base_attributes.merge(inventory_quantity: inventory_quantity_for(product[:product_id]))
+            product_with_base_attributes
+              .merge(inventory_quantity: inventory_quantity_for(product[:product_id]))
+              .merge(variant_option_values(product_with_base_attributes))
+          when 'configurable'
+            product_with_base_attributes
+              .merge(configurable_product_options(product_with_base_attributes))
           else
             product_with_base_attributes
           end
         end
 
-        def info_for(product_id)
+        def variant_option_values(simple_product)
+          @product_options.shopify_variant_options(simple_product)
+        end
+
+        def configurable_product_options(product)
+          @product_options.shopify_option_names(product[:product_id])
+        end
+
+        def info_for(product)
+          additional_attributes = if product[:parent_id]
+            @product_options.option_names_for_soap(product[:parent_id])
+          end
+
+          attributes = if additional_attributes
+            { 'additional_attributes' => { item: additional_attributes } }
```

Comment:
Any reason why this is nested?

Reply:
I thought without the block the variable will be undefined. I wanted it to be nil if the condition failed.

But I was wrong, I'll change it.

```
irb(main):001:0> a = 5 if false
=> nil
irb(main):002:0> a
=> nil
```